sgkit/model.py

```diff
@@ -15,43 +16,46 @@
 def create_genotype_call_dataset(
     *,
     variant_contig_names: List[str],
-    variant_contig: Any,
-    variant_position: Any,
-    variant_alleles: Any,
-    sample_id: Any,
-    call_genotype: Any,
-    call_genotype_phased: Any = None,
-    variant_id: Any = None,
+    variant_contig: ArrayLike,
+    variant_position: ArrayLike,
+    variant_alleles: ArrayLike,
+    sample_id: ArrayLike,
+    call_genotype: ArrayLike,
+    call_genotype_phased: Optional[ArrayLike] = None,
```

Comment:
I noticed that this is shown in the rendered doc as `Union[ndarray, Array, None]`. Not necessarily a problem, but it means we probably should keep the "optional" marker in the description (as you have done).

Reply:
Yeah, I was thinking its even better that it shows explicitly what array like mean.